src/fullerite/collector/docker_stats.go

```
@@ -308,3 +387,140 @@ func min(x, y int) int {
+            devNameMinMajMap[rec[2]] = []string{rec[0], rec[1], rec[2], rec[3], rec[4], rec[5], rec[6], rec[7], rec[8], rec[9], rec[10], rec[11], rec[12], rec[13]}
+       }
+    }
+
+    if err != io.EOF {
+        fmt.Printf(" > Failed!: %v\n", err)
+    }
+
+    return devNameMinMajMap, err
+}
+
+// returns a collection of records each having (device name, major, minor, mount path, reads, writes)
+func ObtainDiskIOStats(fn string, diskStats map[string][]string) ([]DiskIOStats, error) {
+    var diskIOStatsList []DiskIOStats
+    var major int
+    var minor int
+    var reads float64
+    var writes float64
+
+    file, err := os.Open(fn)
+    defer file.Close()
+
+    if err != nil {
+        return diskIOStatsList, err
+    }
+
+    // read the contents of the file with a reader
+    reader := bufio.NewReader(file)
+
+    // read line-by-line
+    var line string
+    for {
+        line, err = reader.ReadString('\n')
+
+        if err != nil {
+            break
+        }
+
+        // split the line on space
+       values := strings.Fields(line)
+       deviceName := values[0]
+
+       // since we could either find an exact match of the device name or /dev/ appended to it in /proc/mounts
+       if(strings.Contains(deviceName, "/")) {
+           // extract the value after the last slash, since that should be the device name
+           splitOnSlash := strings.Split(values[0], "/")
+           deviceName = splitOnSlash[len(splitOnSlash)-1]
+       }
+       // check if the deviceName is present in the devNameMinMaj map
+       if _, ok := diskStats[deviceName]; ok {
+           // extract the major and minor values
+           major, _ = strconv.Atoi(diskStats[deviceName][0])
+           minor, _ = strconv.Atoi(diskStats[deviceName][1])
+           reads, _ = strconv.ParseFloat(diskStats[deviceName][3], 64)
+           writes, _ = strconv.ParseFloat(diskStats[deviceName][7], 64)
```

Please don't ignore errors